tests/integration/test_integration.py

```diff
@@ -583,17 +583,16 @@ def encode_traces(self, traces):
 
 
 @skip_if_testagent
-@parametrize_with_all_encodings(err=None)
+@pytest.mark.subprocess(err=None)
 def test_api_version_downgrade_generates_no_warning_logs():
-    import os
-
     import mock
 
     from ddtrace import tracer as t
+    from ddtrace.internal.utils.http import Response
 
-    encoding = os.environ["DD_TRACE_API_VERSION"] or "v0.5"
-    t._writer._downgrade(None, None, t._writer._clients[0])
-    assert t._writer._endpoint == {"v0.5": "v0.4/traces", "v0.4": "v0.3/traces"}[encoding]
+    t._writer.api_version = "v0.5"
```

Comment:
Why this change?

Reply:
The downgrade test currently validates updating the trace api from v0.5 to v0.4 and v0.4 to v0,3. Since we are dropping support for v0.3 endpoint we only want to test downgrading the trace api from v0.5 to v0.4.